src/main/java/com/example/domain/User.java

```diff
@@ -0,0 +1,38 @@
+package com.example.domain;
+
+public class User {
+	private String username;
+	private String password;
+	public enum Role {ADMIN, USER};
```

Comment:
You don't need ";" after this declaration. Additionally prefer putting nested classes/enums after the methods of enclosing class. If your nested enum had some methods it would be right between User's fields and methods and would look messy.